File: Homework/Son/W2/1636-sort-array-by-increasing-frequency.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
 var frequencySort = function(nums) {
    let map = new Map();
    for (let n of nums) {
      map.set(n, map.get(n) + 1 || 1);
    }
    return nums.sort((a, b) =>
      map.get(a) === map.get(b) ? b - a : map.get(a) - map.get(b)
    );
  };

//   https://leetcode.com/problems/sort-array-by-increasing-frequency/

File: Homework/Ngan_Luu/1636-sort-array-by-increasing-frequency.py
https://leetcode.com/.../sort-array-by-increasing-frequency/
class Solution:
    def frequencySort(self, nums: List[int]) -> List[int]:
        count={}
        for number in nums:
            if number in count:
                count[number].append(number)
            else:
                count[number]=[number]
        count_in_list=[None]*201
        for key,value in reversed(sorted(count.items())):
            if count_in_list[len(value)] is None:
                count_in_list[len(value)]=value
            else:
                count_in_list[len(value)]=count_in_list[len(value)]+value
        output=[]
        for fre in count_in_list:
            if fre is not None:
                output+=fre
        return output